### PR TITLE
Add Transaction isolation level to build step

### PR DESCRIPTION
**Pull request by PeterGerrard**
Add transaction isolation level as a settable option in the build step.

Tested upgrade, and setting the value to ones with and without spaces.

### CODE REVIEW

File: src/main/java/redgatesqlci/BuildBuilder.java [hunk 1]
```diff
@@ -201,6 +209,11 @@ public boolean perform(final AbstractBuild build, final Launcher launcher, final
             params.add(getEscapedOptions(getOptions()));
         }
 
+        if (getTransactionIsolationLevel() != null) {
```
Comment (IvoMiller) [inline src/main/java/redgatesqlci/BuildBuilder.java]: I'd just access the final field directly

File: src/main/java/redgatesqlci/BuildBuilder.java [hunk 2]
```diff
@@ -303,6 +304,14 @@ public FormValidation doCheckPackageid(@QueryParameter final String value) {
             return FormValidation.ok();
         }
 
+        public ListBoxModel doFillTransactionIsolationLevelItems() {
+            ListBoxModel items = new ListBoxModel();
```
Comment (IvoMiller) [inline src/main/java/redgatesqlci/BuildBuilder.java]: I'd lambda this

File: src/main/java/redgatesqlci/TransactionIsolationLevel.java [hunk 3]
```diff
@@ -1,5 +1,21 @@
 package redgatesqlci;
 
+import java.util.Optional;
+
 public enum TransactionIsolationLevel {
-    Serializable, Snapshot, RepeatableRead, ReadCommitted, ReadUncommitted
+    Serializable(), Snapshot(), RepeatableRead("Repeatable Read"), ReadCommitted("Read Committed"), ReadUncommitted("Read Uncommitted");
+
+    private final Optional<String> displayName;
```
Comment (IvoMiller) [inline src/main/java/redgatesqlci/TransactionIsolationLevel.java]: Would just make this required

**Review by IvoMiller — APPROVED** [review]
Happy with the overall change. Just a couple of cosmetic tweaks requested.

**Comment by lubosmichalic** [comment]
Looks allright, following up on having some Java syntax explained.

**Pull request merged**